### DISCUSSION TITLE
Encrypt the password manually

### DISCUSSION THREAD

**Discussion by JustLooKeD** [rustdesk/rustdesk#4979] [Q&A]
Good time of day. After the recent update to the latest version of RustDesk for Linux, I noticed that RustDesk now hashes/encrypts (not sure which one exactly) the permanent password and ID in the user settings file (~/.config/rustdesk/RustDesk.toml). How can I manually encrypt or hash the permanent password in order to insert it into the 'password' parameter? I need this to automate the installation of RustDesk using Puppet. Also, is it possible to retrieve the computer ID? The 'enc_id' parameter was not encrypted before, which allowed me to gather facts about the computer ID using Puppet. Is there a way to decrypt it or somehow retrieve it using a command?

**Comment by dinger1986** [comment]
You can set the password using `--password randompassword` and get the ID using `--get-id`

**Reply by JustLooKeD** [reply]
When I try to use RustDesk in terminal with these arguments, I get an error that graphics are required '(rustdesk:3824): Gtk-WARNING **: 10:39:37.434: cannot open display:

**Comment by mitenpetal** [comment]
I want to know how to get the password in straight text. what hash does it use ?
